src/main/kotlin/com/bloock/sdk/record/entity/Record.kt

```diff
@@ -100,13 +99,21 @@ class Record<T>(private val hash: String, private val document: Document<T>? = n
         if (signatures?.isNotEmpty() == true) {
             return this.signing.verify(this.document?.getSignatures(), signatures)
         } else {
-           //TODO throw NoSignatureFoundException()
+            //TODO throw NoSignatureFoundException()
             return true
         }
     }
 
     fun setProof(proof: Proof) = this.document?.setProof(proof) ?: Unit
-    fun getProof(proof: Proof) = this.document?.getProof() ?: Unit
+    fun getProof(): Proof? {
+        return ifIsDocument()
+    }
+
+    private fun ifIsDocument(): Proof? {
```

Comment:
Why you separate with another function ifIsDocument() ?

Reply:
I thought I'd use it elsewhere in the code, but I forgot to refactor it again